core/domain/base_model_validators.py

```diff
@@ -351,26 +350,33 @@ def _fetch_field_name_to_external_model_references(cls, item):
         """
         multiple_models_ids_to_fetch = {}
 
-        for external_model_fetcher_details in (
-                cls._get_external_id_relationships(item)):
-            multiple_models_ids_to_fetch[
-                external_model_fetcher_details.field_name] = (
-                    external_model_fetcher_details.model_class,
-                    external_model_fetcher_details.model_ids)
-
-        fetched_model_instances_for_all_ids = (
-            datastore_services.fetch_multiple_entities_by_ids_and_models(
-                list(multiple_models_ids_to_fetch.values())))
-
-        for index, field_name in enumerate(multiple_models_ids_to_fetch):
-            (model_class, model_ids) = multiple_models_ids_to_fetch[field_name]
-            fetched_model_instances = fetched_model_instances_for_all_ids[index]
-
-            for (model_id, model_instance) in python_utils.ZIP(
-                    model_ids, fetched_model_instances):
-                cls.field_name_to_external_model_references[field_name].append(
-                    ExternalModelReference(
-                        model_class, model_id, model_instance))
+        try:
+            for external_model_fetcher_details in (
+                    cls._get_external_id_relationships(item)):
+                multiple_models_ids_to_fetch[
+                    external_model_fetcher_details.field_name] = (
+                        external_model_fetcher_details.model_class,
+                        external_model_fetcher_details.model_ids)
+
+            fetched_model_instances_for_all_ids = (
+                datastore_services.fetch_multiple_entities_by_ids_and_models(
+                    list(multiple_models_ids_to_fetch.values())))
+
+            for index, field_name in enumerate(multiple_models_ids_to_fetch):
+                (model_class, model_ids) = (
+                    multiple_models_ids_to_fetch[field_name])
+                fetched_model_instances = fetched_model_instances_for_all_ids[
+                    index]
+
+                for (model_id, model_instance) in python_utils.ZIP(
+                        model_ids, fetched_model_instances):
+                    cls.field_name_to_external_model_references[
+                        field_name].append(
+                            ExternalModelReference(
+                                model_class, model_id, model_instance))
+
+        except utils.ValidationError as err:
+            cls._add_error(ERROR_CATEGORY_INVALID_USER_SETTING_IDS, err)
```

Comment:
I would maybe do this instead, so that the try is on the least amount of lines.
```suggestion
        try:
            for external_model_fetcher_details in (
                    cls._get_external_id_relationships(item)):
                multiple_models_ids_to_fetch[
                    external_model_fetcher_details.field_name] = (
                        external_model_fetcher_details.model_class,
                        external_model_fetcher_details.model_ids)
        except utils.ValidationError as err:
            cls._add_error(ERROR_CATEGORY_INVALID_USER_SETTING_IDS, err)
		else:
            fetched_model_instances_for_all_ids = (
                datastore_services.fetch_multiple_entities_by_ids_and_models(
                    list(multiple_models_ids_to_fetch.values())))

            for index, field_name in enumerate(multiple_models_ids_to_fetch):
                (model_class, model_ids) = (
                    multiple_models_ids_to_fetch[field_name])
                fetched_model_instances = fetched_model_instances_for_all_ids[
                    index]

                for (model_id, model_instance) in python_utils.ZIP(
                        model_ids, fetched_model_instances):
                    cls.field_name_to_external_model_references[
                        field_name].append(
                            ExternalModelReference(
                                model_class, model_id, model_instance))
```

Reply:
Done.